File: ui/src/types/favorites.ts
/**
 * Type definitions for favorites API responses
 * Matches the backend API structure from quizninja-api
 */

export interface QuizSummary {
  id: string;
  title: string;
  description: string;
  category: string;
  difficulty: string;
  time_limit: number;
  question_count: number;
  points: number;
  is_featured: boolean;
  tags: string[];
  thumbnail_url?: string;
  created_at: string;
}

export interface UserQuizFavorite {
  id: string;
  user_id: string;
  quiz_id: string;
  favorited_at: string;
  quiz: QuizSummary;
}

export interface FavoritesListResponse {
  favorites: UserQuizFavorite[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

/**
 * API response wrapper
 * The backend wraps all responses in a "data" field
 */
export interface FavoritesApiResponse {
  data: FavoritesListResponse;
}

File: api/models/favorites.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// UserQuizFavorite represents a user's favorite quiz
type UserQuizFavorite struct {
	ID          uuid.UUID    `json:"id" db:"id"`
	UserID      uuid.UUID    `json:"user_id" db:"user_id"`
	QuizID      uuid.UUID    `json:"quiz_id" db:"quiz_id"`
	FavoritedAt time.Time    `json:"favorited_at" db:"favorited_at"`
	Quiz        *QuizSummary `json:"quiz,omitempty"`
}

// FavoritesListResponse represents the response for user favorites
type FavoritesListResponse struct {
	Favorites  []UserQuizFavorite `json:"favorites"`
	Total      int                `json:"total"`
	Page       int                `json:"page"`
	PageSize   int                `json:"page_size"`
	TotalPages int                `json:"total_pages"`
}

// AddFavoriteRequest represents the request to add a quiz to favorites
type AddFavoriteRequest struct {
	QuizID uuid.UUID `json:"quiz_id" binding:"required"`
}